subsys/net/lib/dhcpv4/dhcpv4.c

```diff
@@ -810,6 +817,68 @@ static bool dhcpv4_parse_options(struct net_pkt *pkt,
 			break;
 		}
 #endif
+
+#if defined(CONFIG_NET_DHCPV4_OPTION_CALLBACKS_VENDOR_SPECIFIC)
+		case DHCPV4_OPTIONS_VENDOR_SPECIFIC: {
+
+			if (length >= 3) {
+
+				struct net_dhcpv4_option_callback *vs_cb, *vs_tmp;
```

Comment:
As there are multiple level of `{}` blocks below, I would suggest to make the added code as a function that we could call here. This would help the readability of the code.